rasa/core/domain.py

```diff
@@ -265,30 +265,61 @@ def collect_slots(slot_dict: Dict[Text, Any]) -> List[Slot]:
 
     @staticmethod
     def collect_intent_properties(
-        intents: List[Union[Text, Dict[Text, Any]]]
+        intents: List[Union[Text, Dict[Text, Any]]], entities: List[Text]
     ) -> Dict[Text, Dict[Text, Union[bool, List]]]:
         intent_properties = {}
         for intent in intents:
-            if isinstance(intent, dict):
-                name = list(intent.keys())[0]
-                for properties in intent.values():
-                    properties.setdefault("use_entities", True)
-                    properties.setdefault("ignore_entities", [])
-                    if (
-                        properties["use_entities"] is None
-                        or properties["use_entities"] is False
-                    ):
-                        properties["use_entities"] = []
-            else:
-                name = intent
+            if not isinstance(intent, dict):
                 intent = {intent: {"use_entities": True, "ignore_entities": []}}
 
+            name, properties = list(intent.items())[0]
+
             if name in intent_properties.keys():
                 raise InvalidDomain(
                     "Intents are not unique! Found two intents with name '{}'. "
                     "Either rename or remove one of them.".format(name)
                 )
 
+            if (
+                properties.get("used_entities") is None
+            ):  # important to differentiate between None and [] here
```

Comment:
One case is that the field exists and contains '[]' (no entities are used), the other case is that the field doesn't exist (because instead there is a field `use_entities` or `ignore_entities`).

But you are right, this allows adding `used_entities: None` to an intent in your `domain.yml`, which produces unwanted behaviour: It would use the default `use_entities: True` and put all entities into `used_entities`.

Reply:
This would also automatically be addressed/solved by what I suggest below (having `as_dict()` return something with property `use_entities` or `ignore_entities` instead of `used_entities`).